crates/anvil/src/eth/api.rs

```diff
@@ -528,11 +528,12 @@ impl EthApi {
     /// Handler for ETH RPC call: `eth_accounts`
     pub fn accounts(&self) -> Result<Vec<Address>> {
         node_info!("eth_accounts");
-        let mut accounts = Vec::new();
+        let mut accounts = HashSet::new();
         for signer in self.signers.iter() {
-            accounts.append(&mut signer.accounts());
+            accounts.extend(signer.accounts());
         }
-        Ok(accounts)
+        accounts.extend(self.backend.cheats().impersonated_accounts());
```

Comment:
I'd like to keep a consistent order here and still use vec, but use an additional hashmap to filter unique addresses,
`accounts.extend(signer.accounts().filter(|acc| unique.insert(*acc)))`

Reply:
I implemented what I **think** you suggested: [869ef4d](https://github.com/foundry-rs/foundry/pull/5734/commits/869ef4d6730804e82adb3fd473e1c29d5ca576b3)
However, this ended up a bit verbose because the filter needs to be applied to all iterators. If you don't care about listing the original accounts first and the impersonated accounts later an alternative implementation could also be this:
```rust
        let mut accounts = Vec::new();
        for signer in self.signers.iter() {
            accounts.append(&mut signer.accounts());
        }
        accounts.extend(self.backend.cheats().impersonated_accounts());
        accounts.sort();
        accounts.dedup();
```
LMK what you prefer.

Reply:
Actually since `impersonated_accounts` already returns a `HashMap` the iterator produced by that already has unpredictable order if you are impersonating > 1 accounts.